lms/djangoapps/course_home_api/outline/v1/serializers.py

```diff
@@ -45,6 +30,29 @@ def get_blocks(self, blocks):
         }
 
 
+class CourseGoalSerializer(serializers.Serializer):
+    """
+    Serializer for Course Goal data
+    """
+    goal_options = serializers.DictField()
```

Comment:
The consuming side of this might also lose ordering when translating the serialization into a dictionary. I don't know if Javascript has order guarantees for objects. A DictField is probably the wrong data type for an ordered collection.